src/library/scala/collection/immutable/Queue.scala

```diff
@@ -108,8 +108,14 @@ sealed class Queue[+A] protected(protected val in: List[A], protected val out: L
 
   override def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Queue[A], B, That]): That = {
     if (bf eq Queue.ReusableCBF) {
-      val thatQueue = that.asInstanceOf[Queue[B]]
-      new Queue[B](thatQueue.in ++ (thatQueue.out reverse_::: this.in), this.out).asInstanceOf[That]
+      val newIn =
+        if (that.isInstanceOf[Queue[_]]) {
+          val thatQueue: Queue[B] = that.asInstanceOf[Queue[B]]
```

Comment:
Yeah but I already asked him to rewrite it once.  I guess it might be more idiomatic if it were a match.  (Personally I don't really like unchecked matches; if one is going to cheat I prefer to `asInstanceOf` it.)